compiler/src/iree/compiler/Dialect/LinalgExt/Transforms/ReshapeFusion.cpp

```diff
@@ -652,6 +652,93 @@ struct DropScatterUnitIndexDepth final : public OpRewritePattern<ScatterOp> {
   }
 };
 
+struct DropGatherBatchUnitDims final : public OpRewritePattern<GatherOp> {
+  using OpRewritePattern<GatherOp>::OpRewritePattern;
+  LogicalResult matchAndRewrite(GatherOp gatherOp,
```

Comment:
I think we can return a failure when the op does not have pure tensor semantics? (You can check it with `gatherOp.hasPureTensorSemantics()`, fyi).